lib/httparty/connection_adapter.rb

```diff
@@ -126,6 +133,14 @@ def connection
         http.open_timeout = options[:open_timeout]
       end
 
+      if options[:write_timeout] && (options[:write_timeout].is_a?(Integer) || options[:write_timeout].is_a?(Float))
```

Comment:
There is a lot of duplication here. Would be better to extract it to a private method `add_timeout?(options[:write_timeout])`